crates/uv-resolver/src/version_map.rs

```diff
@@ -138,24 +146,50 @@ impl VersionMap {
     /// for each version.
     pub(crate) fn iter(
         &self,
+        range: &Range<Version>,
     ) -> impl DoubleEndedIterator<Item = (&Version, VersionMapDistHandle)> + ExactSizeIterator {
-        match self.inner {
-            VersionMapInner::Eager(ref map) => {
-                either::Either::Left(map.iter().map(|(version, dist)| {
-                    let version_map_dist = VersionMapDistHandle {
-                        inner: VersionMapDistHandleInner::Eager(dist),
-                    };
-                    (version, version_map_dist)
-                }))
-            }
-            VersionMapInner::Lazy(ref lazy) => {
-                either::Either::Right(lazy.map.iter().map(|(version, dist)| {
-                    let version_map_dist = VersionMapDistHandle {
-                        inner: VersionMapDistHandleInner::Lazy { lazy, dist },
-                    };
-                    (version, version_map_dist)
-                }))
-            }
+        if let Some(version) = range.as_singleton() {
```

Comment:
```suggestion
        // Performance optimization: If we only have a single version, return that version directly.
        if let Some(version) = range.as_singleton() {
```

Without a comment, i can already see myself looking at this code later and wondering whether the branch is load-bearing or not.